File: Backend/carShop/src/main/java/com/contrader/carShop/dto/AnagraficaDTO.java
package com.contrader.carShop.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.contrader.carShop.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
public class AnagraficaDTO {

    private Long id;

    private String nome;

    private String cognome;

    private String genere;

    private String nazione;

    private String provincia;

    private String cittaResidenza;

    private String indirizzo;

    private String dataNascita;

    private User user;
}

File: Frontend/src/dto/anagraficadto.ts
import { UserDTO } from "./userdto";

export class AnagraficaDTO{
    id: number;
    nome: String;
    cognome: String;
    genere: String;
    nazione: String;
    provincia: String;
    cittaResidenza: String;
    indirizzo: String;
    dataNascita: String;
    user: UserDTO;

    constructor(id: number,
                nome: String,
                cognome: String,
                genere: String,
                nazione: String,
                provincia: String,
                cittaResidenza: String,
                indirizzo: String,
                dataNascita: String,
                user: UserDTO){
        this.id = id;
        this.nome = nome;
        this.cognome = cognome;
        this.genere = genere;
        this.nazione = nazione;
        this.provincia = provincia;
        this.cittaResidenza = cittaResidenza;
        this.indirizzo = indirizzo;
        this.dataNascita = dataNascita;
        this.user = user;
    }
}